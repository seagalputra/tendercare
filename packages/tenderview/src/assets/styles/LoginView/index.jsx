import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(8),
    [theme.breakpoints.down('xs')]: {
      marginTop: theme.spacing(0)
    },
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: theme.spacing(6)
  },
  title: {
    textAlign: 'center',
    fontWeight: 'bold'
  },
  form: {
    width: '100%',
    marginTop: theme.spacing(1)
  },
  submit: {
    margin: theme.spacing(3, 0, 2)
  },
  image: {
    width: '78%',
    margin: 'auto'
  }
}))

export default useStyles
