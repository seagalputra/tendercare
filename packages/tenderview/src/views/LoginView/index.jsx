import React, { useState } from 'react'
import {
  Button,
  TextField,
  FormControlLabel,
  Checkbox,
  Link,
  Grid,
  Typography,
  Container,
  Box,
  CircularProgress
} from '@material-ui/core'
import { useFormik } from 'formik'
import { useHistory } from 'react-router-dom'
import * as Yup from 'yup'

import authenticateUser from 'api/user'
import { storeToken } from 'utils/sessions'

import useStyles from 'assets/styles/LoginView'

const validationForm = Yup.object({
  email: Yup.string()
    .email('Masukkan alamat email yang valid')
    .required(),
  password: Yup.string().required()
})

const LoginView = () => {
  const [isLoading, setIsLoading] = useState(false)
  const classes = useStyles()
  const history = useHistory()

  const handleRedirect = path => {
    history.push(path)
  }

  const handleLogin = async request => {
    const { email, password } = request
    setIsLoading(true)
    const {
      data: { token }
    } = await authenticateUser(email, password)
    storeToken(token)
    handleRedirect('/dashboard')
  }

  const formik = useFormik({
    initialValues: {
      email: '',
      password: ''
    },
    validationSchema: validationForm,
    onSubmit: data => handleLogin(data)
  })

  return (
    <Container component="main" maxWidth="sm">
      <Box className={classes.paper}>
        <Typography className={classes.title} variant="h5" gutterBottom>
          Selamat Datang di SPO 1047
        </Typography>
        <Typography variant="subtitle1">
          Silahkan masuk untuk melanjutkan
        </Typography>

        <form className={classes.form} onSubmit={formik.handleSubmit}>
          <TextField
            error={formik.touched.email && Boolean(formik.errors.email)}
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="email"
            label="Alamat Email"
            name="email"
            autoComplete="email"
            type="email"
            autoFocus
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            value={formik.values.email}
          />
          <TextField
            error={formik.touched.password && Boolean(formik.errors.password)}
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="password"
            label="Kata Sandi"
            name="password"
            type="password"
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            value={formik.values.password}
          />
          <FormControlLabel
            control={<Checkbox value="remember" color="primary" />}
            label="Ingat Saya"
          />
          {isLoading ? (
            <Button
              fullWidth
              variant="contained"
              disabled
              className={classes.submit}
            >
              <CircularProgress size={24} color="primary" />
            </Button>
          ) : (
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Masuk
            </Button>
          )}
          <Grid container>
            <Grid item xs>
              <Link href="!#" variant="body2">
                Lupa Password?
              </Link>
            </Grid>
            <Grid item>
              <Link href="!#" variant="body2">
                Tidak punya akun? Daftar
              </Link>
            </Grid>
          </Grid>
        </form>
      </Box>
    </Container>
  )
}

export default LoginView
