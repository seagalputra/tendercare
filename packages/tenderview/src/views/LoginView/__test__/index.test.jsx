import React from 'react'
import { shallow } from 'enzyme'
import LoginView from '..'

describe('LoginView', () => {
  it('renders without crashing', () => {
    const component = shallow(<LoginView />)
    expect(component).toMatchSnapshot()
  })
})
