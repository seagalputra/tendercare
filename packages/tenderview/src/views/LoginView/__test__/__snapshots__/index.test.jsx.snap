// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView renders without crashing 1`] = `
<WithStyles(ForwardRef(Grid))
  component="main"
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    item={true}
    justify="center"
    md={7}
    sm={5}
    xs={false}
  >
    <img
      alt="Background"
      className="makeStyles-image-5"
      src="delivery.svg"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-paper-1"
    item={true}
    md={5}
    sm={7}
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      gutterBottom={true}
      variant="h5"
    >
      Selamat Datang di SPO 1047
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="subtitle1"
    >
      Silahkan masuk untuk melanjutkan
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-3"
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="email"
        autoFocus={true}
        fullWidth={true}
        id="email"
        label="Alamat Email"
        margin="normal"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="email"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        id="password"
        label="Kata Sandi"
        margin="normal"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="password"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(FormControlLabel))
        control={
          <ForwardRef(WithStyles)
            color="primary"
            value="remember"
          />
        }
        label="Ingat Saya"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-4"
        color="primary"
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        Masuk
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        >
          <WithStyles(ForwardRef(Link))
            href="!#"
            variant="body2"
          >
            Lupa Password?
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Link))
            href="!#"
            variant="body2"
          >
            Tidak punya akun? Daftar
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
